modules/axhal/src/arch/riscv/context.rs

```diff
@@ -53,6 +53,136 @@ pub struct TrapFrame {
     pub sstatus: usize,
 }
 
+impl TrapFrame {
```

Comment:
I really think this should be written as macros
some thing like 
```Rust
macro_rules! define_syscall_args {
    ($struct_name:ident, $( $index:expr ),*) => {
        impl $struct_name {
            $( 
                /// Gets the syscall argument.
                pub const fn arg$index(&self) -> usize {
                    self.regs.a$index
                }
            )*
        }
    };
}

// Usage
define_syscall_args!(TrapFrame, 0, 1, 2, 3, 4, 5);
```

Reply:
This won't work due to rust's macro rules. To create a identifier by concatenation, we must use a proc_macro like `paste::paste!`, `arg$index` just won't work. But I do agree that we need a better way to declare syscall arg accessors.

Reply:
We can do this without separating the register names in the middle：

```rs
define_syscall_args!(TrapFrame, rdi, rsi, rdx, r10, r8, r9); // x86_64
define_syscall_args!(TrapFrame, regs.a0, regs.a1, regs.a2, regs.a3, regs.a4, regs.a5); // riscv64
define_syscall_args!(TrapFrame, r[0], r[1], r[2], r[3], r[4], r[5]); // aarch64
```

Reply:
It may be in the next PR, let's merge this first.